x-pack/solutions/security/plugins/cloud_security_posture/public/plugin.tsx

```diff
@@ -139,6 +167,24 @@ export class CspPlugin
           ),
         };
       },
+      getCloudSecurityPostureVulnerabilityFlyout: () => {
+        return {
+          Component: (props: FindingVulnerabilityFlyoutProps) => (
+            <LazyCspFindingsVulnerabilityFlyout {...props}>
+              {props.children}
```

Comment:
Just curious, what the reason behind lazy loading these components?

Reply:
Previously what we are planning is basically to move the components to Security Solution plugins, but after consideration and checking it out, due to how entangled our components are (1 is connected to multiple components etc) we decided that would be too much to do 

By splitting and lazy loading the flyout into multiple parts like we are able to just call this component in Security Solution plugin without the need of moving the component files, furthermore it also allows us to wrap each part using certain component that might only exist in Security Solution plugin

@opauloh

Reply:
> By splitting and lazy loading the flyout into multiple parts like we are able to just call this component in Security Solution plugin without the need of moving the component files...

Yes, and by exporting a lazy-loaded component, it means that the bundle size of the Security Solution plugin will not increase by just importing and starting the Cloud Security plugin. In the real world scenario means the browser won't load assets before the user needs it (i.e, it only downloads the JavaScript related to the Cloud Security posture components when the user opens a flyout with Cloud Security posture data)

Reply:
Okay got it! Thanks for the context!